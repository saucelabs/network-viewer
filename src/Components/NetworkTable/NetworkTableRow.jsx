import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';

import { VIEWER_FIELDS, ROW_ID_PREFIX } from './../../constants';
import Styles from './NetworkTableHeader.styles.scss';
import TimeChart from './TimeChart';
import NetworkCellValue from './NetworkCellValue';
import { getStatusClass } from '../../utils';

const context = classNames.bind(Styles);

const NetworkTableRow = ({
  payload,
  maxTime,
  scrollHighlight,
  onSelect,
}) => {
  const handleSelectRequest = () => {
    onSelect(payload);
  };

  const rowProps = {
    className: context(
      'network-table-row',
      getStatusClass(payload), {
        highlight: scrollHighlight,
      }),
    id: ROW_ID_PREFIX + payload.index,
    onClick: handleSelectRequest,
  };

  return (
    <tr {...rowProps}>
      {Object.entries(VIEWER_FIELDS).map(([datakey, { key, unit }]) => (
        <NetworkCellValue
          key={datakey}
          datakey={key}
          payload={payload}
          unit={unit}
        />
      ))}
      <td className={Styles['timeline-header']}>
        {payload.time ? (
          <TimeChart
            maxTime={maxTime}
            timings={payload.timings}
          />
        ) : ''}
      </td>
    </tr>
  );
};

NetworkTableRow.propTypes = {
  maxTime: PropTypes.number.isRequired,
  onSelect: PropTypes.func.isRequired,
  payload: PropTypes.object.isRequired,
  scrollHighlight: PropTypes.bool.isRequired,
};

export default NetworkTableRow;
