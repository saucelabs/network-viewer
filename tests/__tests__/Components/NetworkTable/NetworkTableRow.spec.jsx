import React from 'react';
import { mount } from 'enzyme';

import NetworkTableRow from './../../../../src/Components/NetworkTable/NetworkTableRow';

describe('NetworkTableRow', () => {
  const NOOP = () => {
  };

  const props = {
    entry: {
      domain: 'developer.mozilla.org',
      filename: 'slice',
      index: 0,
      method: 'GET',
      size: 135.04,
      startedDateTime: '2019-10-14T08:47:11.592Z',
      status: 200,
      timings: {
        _blocked_queueing: 4.382999992230907,
        blocked: 5.496999992230907,
        connect: -1,
        dns: -1,
        receive: 3.252000024076551,
        send: 0.2819999999999998,
        ssl: -1,
        startTime: 0,
        wait: 26.03899999633059,
      },
      type: 'html',
      url: 'https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice',
    },
    maxTime: 5000,
    scrollHighlight: false,
    onSelect: NOOP,
  };

  it('renders without crashing', () => {
    const element = mount(
      <table>
        <tbody>
          <NetworkTableRow {...props} />
        </tbody>
      </table>,
    );
    expect(element)
      .toMatchSnapshot();
  });
});
