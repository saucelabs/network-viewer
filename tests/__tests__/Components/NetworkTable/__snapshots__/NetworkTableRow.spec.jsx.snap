// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NetworkTableRow renders without crashing 1`] = `
<table>
  <tbody>
    <NetworkTableRow
      maxTime={5000}
      onSelect={[Function]}
      payload={
        Object {
          "domain": "developer.mozilla.org",
          "filename": "slice",
          "index": 0,
          "method": "GET",
          "size": 135.04,
          "startedDateTime": "2019-10-14T08:47:11.592Z",
          "status": 200,
          "timings": Object {
            "_blocked_queueing": 4.382999992230907,
            "blocked": 5.496999992230907,
            "connect": -1,
            "dns": -1,
            "receive": 3.252000024076551,
            "send": 0.2819999999999998,
            "ssl": -1,
            "startTime": 0,
            "wait": 26.03899999633059,
          },
          "type": "html",
          "url": "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice",
        }
      }
      scrollHighlight={false}
    >
      <tr
        className="network-table-row info"
        id="network-viewer-table-row-0"
        onClick={[Function]}
      >
        <NetworkCellValue
          datakey="filename"
          key="file"
          payload={
            Object {
              "domain": "developer.mozilla.org",
              "filename": "slice",
              "index": 0,
              "method": "GET",
              "size": 135.04,
              "startedDateTime": "2019-10-14T08:47:11.592Z",
              "status": 200,
              "timings": Object {
                "_blocked_queueing": 4.382999992230907,
                "blocked": 5.496999992230907,
                "connect": -1,
                "dns": -1,
                "receive": 3.252000024076551,
                "send": 0.2819999999999998,
                "ssl": -1,
                "startTime": 0,
                "wait": 26.03899999633059,
              },
              "type": "html",
              "url": "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice",
            }
          }
          unit={null}
        >
          <td
            className="value-cell filename"
          >
            <Tooltip
              delay={500}
              placement="top"
              title="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice"
            >
              <span
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onPointerDown={[Function]}
                onTouchStart={[Function]}
              >
                slice
              </span>
              <TooltipTransition
                isOpen={false}
              />
            </Tooltip>
          </td>
        </NetworkCellValue>
        <NetworkCellValue
          datakey="status"
          key="status"
          payload={
            Object {
              "domain": "developer.mozilla.org",
              "filename": "slice",
              "index": 0,
              "method": "GET",
              "size": 135.04,
              "startedDateTime": "2019-10-14T08:47:11.592Z",
              "status": 200,
              "timings": Object {
                "_blocked_queueing": 4.382999992230907,
                "blocked": 5.496999992230907,
                "connect": -1,
                "dns": -1,
                "receive": 3.252000024076551,
                "send": 0.2819999999999998,
                "ssl": -1,
                "startTime": 0,
                "wait": 26.03899999633059,
              },
              "type": "html",
              "url": "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice",
            }
          }
          unit={null}
        >
          <td
            className="value-cell status"
          >
            <span>
              200
            </span>
          </td>
        </NetworkCellValue>
        <NetworkCellValue
          datakey="method"
          key="method"
          payload={
            Object {
              "domain": "developer.mozilla.org",
              "filename": "slice",
              "index": 0,
              "method": "GET",
              "size": 135.04,
              "startedDateTime": "2019-10-14T08:47:11.592Z",
              "status": 200,
              "timings": Object {
                "_blocked_queueing": 4.382999992230907,
                "blocked": 5.496999992230907,
                "connect": -1,
                "dns": -1,
                "receive": 3.252000024076551,
                "send": 0.2819999999999998,
                "ssl": -1,
                "startTime": 0,
                "wait": 26.03899999633059,
              },
              "type": "html",
              "url": "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice",
            }
          }
          unit={null}
        >
          <td
            className="value-cell method"
          >
            <span>
              GET
            </span>
          </td>
        </NetworkCellValue>
        <NetworkCellValue
          datakey="domain"
          key="domain"
          payload={
            Object {
              "domain": "developer.mozilla.org",
              "filename": "slice",
              "index": 0,
              "method": "GET",
              "size": 135.04,
              "startedDateTime": "2019-10-14T08:47:11.592Z",
              "status": 200,
              "timings": Object {
                "_blocked_queueing": 4.382999992230907,
                "blocked": 5.496999992230907,
                "connect": -1,
                "dns": -1,
                "receive": 3.252000024076551,
                "send": 0.2819999999999998,
                "ssl": -1,
                "startTime": 0,
                "wait": 26.03899999633059,
              },
              "type": "html",
              "url": "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice",
            }
          }
          unit={null}
        >
          <td
            className="value-cell domain"
          >
            <span>
              developer.mozilla.org
            </span>
          </td>
        </NetworkCellValue>
        <NetworkCellValue
          datakey="type"
          key="type"
          payload={
            Object {
              "domain": "developer.mozilla.org",
              "filename": "slice",
              "index": 0,
              "method": "GET",
              "size": 135.04,
              "startedDateTime": "2019-10-14T08:47:11.592Z",
              "status": 200,
              "timings": Object {
                "_blocked_queueing": 4.382999992230907,
                "blocked": 5.496999992230907,
                "connect": -1,
                "dns": -1,
                "receive": 3.252000024076551,
                "send": 0.2819999999999998,
                "ssl": -1,
                "startTime": 0,
                "wait": 26.03899999633059,
              },
              "type": "html",
              "url": "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice",
            }
          }
          unit={null}
        >
          <td
            className="value-cell type"
          >
            <span>
              html
            </span>
          </td>
        </NetworkCellValue>
        <NetworkCellValue
          datakey="size"
          key="size"
          payload={
            Object {
              "domain": "developer.mozilla.org",
              "filename": "slice",
              "index": 0,
              "method": "GET",
              "size": 135.04,
              "startedDateTime": "2019-10-14T08:47:11.592Z",
              "status": 200,
              "timings": Object {
                "_blocked_queueing": 4.382999992230907,
                "blocked": 5.496999992230907,
                "connect": -1,
                "dns": -1,
                "receive": 3.252000024076551,
                "send": 0.2819999999999998,
                "ssl": -1,
                "startTime": 0,
                "wait": 26.03899999633059,
              },
              "type": "html",
              "url": "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice",
            }
          }
          unit={null}
        >
          <td
            className="value-cell size"
          >
            <span>
              135.04
            </span>
          </td>
        </NetworkCellValue>
        <NetworkCellValue
          datakey="time"
          key="time"
          payload={
            Object {
              "domain": "developer.mozilla.org",
              "filename": "slice",
              "index": 0,
              "method": "GET",
              "size": 135.04,
              "startedDateTime": "2019-10-14T08:47:11.592Z",
              "status": 200,
              "timings": Object {
                "_blocked_queueing": 4.382999992230907,
                "blocked": 5.496999992230907,
                "connect": -1,
                "dns": -1,
                "receive": 3.252000024076551,
                "send": 0.2819999999999998,
                "ssl": -1,
                "startTime": 0,
                "wait": 26.03899999633059,
              },
              "type": "html",
              "url": "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice",
            }
          }
          unit={null}
        >
          <td
            className="value-cell time"
          >
            <span />
          </td>
        </NetworkCellValue>
        <td />
      </tr>
    </NetworkTableRow>
  </tbody>
</table>
`;
